deps/common/src/logseq/common/path.cljs

```diff
@@ -0,0 +1,302 @@
+(ns logseq.common.path
```

Comment:
I think it's great to have a new library for shared common code. One thing that I'm finding confusing is that this is the 3rd major approach to file and path handling after `frontend.util/node-path.*` and the npm path library. Without some documentation it's unclear when to use which approach. Could you write some docs on when to use this ns? Some questions that would be helpful to answer in docs:
- Is there a plan to deprecate the other approaches and if so, what is it?
- When to use `filename` vs `util/node-path.basename`?
- When to use `file-ext` vs `gp-util/path->file-ext`?
- When to use `path-join` vs `util/node-path.join` or `node-path/join`?
- Is `relative-path` mostly for mobile since it's replacing `capacitor-fs/replacing normalize-file-protocol-path`?